src/ServiceProvider.php

```diff
@@ -67,4 +73,19 @@ protected function getAwsClientClosure(string $client, array $config = [])
             return $app->make('aws')->createClient($client, $config);
         };
     }
+
+    private function getLaravelEnvProvider(): callable
+    {
+        return function () {
+            $key = env(CredentialProvider::ENV_KEY);
```

Comment:
I just realised you won't be able to do it this way. If users of this package are caching configuration, then `env()` will be returning null for any value. So you'll have to put these values into a config file and load them with `config()`

Reply:
I'm confused. The existing provider was using `getenv()` which wouldn't have cared about the caching. Unless laravel clears the $_ENV variable. I don't remember